packages/integration-sdk-runtime/src/logger/index.ts

```diff
@@ -40,6 +40,7 @@ interface CreateLoggerInput<
   invocationConfig?: InvocationConfig<TExecutionContext, TStepExecutionContext>;
   pretty?: boolean;
   serializers?: Logger.Serializers;
+  onFailure?: OnFailureFunction;
```

Comment:
Will this be considered a fatal error? If so, we should consider renaming this to something like `onFatalFailure` for clarity.

Reply:
No, this is for non-fatal failures. Fatal errors terminate the process and throw an error in the calling process, which is currently being handled just fine by the managed integration runtime. It's non-fatal errors that are handled within this package which drives the need for this callback.